Ajout de la DataClass Account, de son repository et controller.

BastienLamour opened the pull request:
Ajout des méthodes nécessaires au CRUD d'un compte utilisateur.

BastienLamour committed:
Ajout de la DataClass Account, de son repository et controller.

BastienLamour commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Comptes](https://trello.com/c/e6IfB5G7/46-comptes)

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/controllers/AccountController.kt:
```diff
@@ -0,0 +1,65 @@
+import org.springframework.web.bind.annotation.*
+
+@RestController
+@RequestMapping("/admin")
```
Pluriel pour le mapping d'un controller

Prakkkmak replied:
J'aurais nommé le mapping accounts

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/controllers/AccountController.kt:
```diff
@@ -0,0 +1,65 @@
+@RequestMapping("/admin")
+class AccountController(private val repository: AccountRepository) {
+    /**
+     * Récupère tous les comptes
```
Ponctuation

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/controllers/AccountController.kt:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    /**
+     * Ajoute un compte
```
Ponctuation

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/controllers/AccountController.kt:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    /**
+     * Modifie le compte en fonction de son identifiant et des nouvelles données
```
Ponctuation

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/controllers/AccountController.kt:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    /**
+     * Supprime le compte en fonction de son identifiant
```
Ponctuation

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/entities/Account.kt:
```diff
@@ -0,0 +1,20 @@
+import org.bson.types.ObjectId
+import org.springframework.data.annotation.Id
+import org.springframework.data.mongodb.core.mapping.Document
+import java.util.*
```
Import * en trop

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/repositories/AccountRepository.kt:
```diff
@@ -0,0 +1,7 @@
+import com.progreizh.animaliste.entities.Account
+import org.springframework.data.mongodb.repository.MongoRepository
+
+interface AccountRepository : MongoRepository<Account, String> {
```
Ajout de findByMail
Ajout de findByMail+Mdp

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/entities/Account.kt:
```diff
@@ -0,0 +1,20 @@
+data class Account(
+    @Id
+    @Generated
+    val id: String = ObjectId.get().toString(),
```
```suggestion
    val id: String = ObjectId.get().toString(),
    val mail: String,
    val password: String,
```

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/entities/Account.kt:
```diff
@@ -0,0 +1,20 @@
+    @Id
+    @Generated
+    val id: String = ObjectId.get().toString(),
+    val name: String,
```
```suggestion
    val name: String?,
```

BastienLamour replied:
J'ai laissé le nom en obligatoire, je pense qu'une vérification de l'identité de façon plus "humaine" serait utile

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/entities/Account.kt:
```diff
@@ -0,0 +1,20 @@
+    @Generated
+    val id: String = ObjectId.get().toString(),
+    val name: String,
+    val surname: String,
+    val contact: String,
```
```suggestion
    val surname: String?,
    val phoneNumber: String?,
```

BastienLamour committed:
Application des suggestions - syntaxe
+ Ajout des methodes findAccountByMailEquals & findAccountByMailEqualsAndPasswordEquals au repository

BastienLamour commented on src/main/kotlin/com/progreizh/animaliste/controllers/AccountController.kt:
```diff
@@ -0,0 +1,79 @@
+            ResponseEntity.ok(accountOptional.get())
+    }
+
+    @GetMapping("/{mail}")
+    fun getOneAccountByMail(@PathVariable("mail") mail: String, @RequestParam password: Optional<String>): ResponseEntity<Account> {
```
Autre possibilité : on prend le nom ou l'id associé à un compte, puis les autres données en paramètres tel que :
`animaliste.fr/dupont?mail=exemple@mail.fr&password=123` ou `animaliste.fr/ID?mail=exemple@mail.fr&password=123` nous retourne le compte voulu.
A discuter.

Prakkkmak replied:
Je connais pas la pratique, Je crois que c'est pas dans l'url mais dans un body je pense :o
Genre /comptes/connect (et en body le mail + psw ) [ avec peut etre un autre nom que connect ]

BastienLamour replied:
Mince j'ai même pas pris le point de vu d'une connexion j'suis débile.
T'as raison du coup pour le body.
La meilleur manière ce serait un body en param et un renvoie d'un booleen plutôt qu'une réponse http ?

Prakkkmak replied:
La réponse HTTP peux etre bonne [https://stackoverflow.com/a/1960453](https://stackoverflow.com/a/1960453)

BastienLamour committed:
Modifs forme

BastienLamour committed:
Modification da méthode getOneAccountByMail en getOneAccountByLoginForm.
Ajout de la data class AccountLoginForm.

BastienLamour committed:
Changement des messages d'erreur pour la méthode getOneAccountByLoginForm

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/repositories/AccountRepository.kt:
```diff
@@ -0,0 +1,10 @@
+import java.util.Optional
+
+interface AccountRepository : MongoRepository<Account, String> {
+    fun findAccountByMailEquals(mail: String): Optional<Account>
```
```suggestion
    fun findAccountByMail(mail: String): Optional<Account>
```
Le nom findAccountByMailEquals est super bizare

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/repositories/AccountRepository.kt:
```diff
@@ -0,0 +1,10 @@
+
+interface AccountRepository : MongoRepository<Account, String> {
+    fun findAccountByMailEquals(mail: String): Optional<Account>
+    fun findAccountByMailEqualsAndPasswordEquals(mail: String, password: String): Optional<Account>
```
```suggestion
    fun findAccountByMailEqualsAndPassword(mail: String, password: String): Optional<Account>
```

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/entities/Account.kt:
```diff
@@ -0,0 +1,22 @@
+    val phoneNumber: String? = null,
+    val legalStatusName: String? = null,
+    val registerDate: Date = Date(),
+
```
```suggestion
```

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/controllers/AccountController.kt:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    /**
+     * Retourne le compte correspondant au couple mail/mot de passe ssi celui-ci existe ou un message d'erreur
```
```suggestion
     * Retourne le compte correspondant au couple mail/mot de passe ssi celui-ci existe ou un message d'erreur.
```

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/entities/AccountLoginForm.kt:
```diff
@@ -0,0 +1,6 @@
+package com.progreizh.animaliste.entities
+
+data class AccountLoginForm(
```
Cette classe me gène, je comprends ton envie de mettre en dehors du body les params de la création d'account. Genre avec l'url /accounts/?mail="blah"password=""
Ca nous éviterai de créer une classe juste pour ça.
Je sais pas trop comment faire :/

BastienLamour replied:
Je remets comme j'avais fait avant du coup, mais avec une route différente (`/accounts/login?mail=x&password=y`).

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/controllers/AccountController.kt:
```diff
@@ -0,0 +1,84 @@
+     * Récupère tous les comptes.
+     */
+    @GetMapping
+    fun getAllAccounts(): ResponseEntity<List<Account>>{
```
```suggestion
    fun getAccounts(): ResponseEntity<List<Account>>{
```
Je sais que dans le controller de l'animal les noms sont pareil mais je pense qu'il faut changer

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/controllers/AccountController.kt:
```diff
@@ -0,0 +1,84 @@
+     * Récupère le compte en fonction de son identifiant.
+     */
+    @GetMapping("/{id}")
+    fun getOneAccountById(@PathVariable("id") id: String): ResponseEntity<Account> {
```
```suggestion
    fun getAccount(@PathVariable("id") id: String): ResponseEntity<Account> {
```
Le prototype de la méthode est assez explicite. Pas besoin de dire byId. De plus , by id est la maniere de base.

Prakkkmak commented on src/main/kotlin/com/progreizh/animaliste/controllers/AccountController.kt:
```diff
@@ -0,0 +1,84 @@
+     * Retourne le compte correspondant au couple mail/mot de passe ssi celui-ci existe ou un message d'erreur
+     */
+    @GetMapping("/login")
+    fun getOneAccountByLoginForm(@RequestBody account: AccountLoginForm): ResponseEntity<Any> {
```
Tu sais si kotlin Spring accepte la surcharge de méthode ?

BastienLamour replied:
J'avais regardé, selon plusieurs topics, je me rappelle plus lesquels, c'est pas possible de surcharger une requête HTTP ou pas conseillé. De toute façon j'ai changé la route donc ça pose pas de pb normalement.

BastienLamour committed:
Récupération d'un compte via le couple mail/password modifié : mise en paramètre de requête HTTP

BastienLamour committed:
Apply suggestions from code review

Co-authored-by: Prakkkmak <Prakkmak@gmail.com>

BastienLamour committed:
Modification valeur retournée pour getAccountByMailAndPassword

BastienLamour committed:
Merge remote-tracking branch 'origin/feature/CRUD-comptes' into feature/CRUD-comptes

BastienLamour committed:
-

BastienLamour committed:
-

BastienLamour committed:
Ajout des TU

BastienLamour commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Jeu de données Comptes et TU](https://trello.com/c/btFRY6KX/48-jeu-de-donn%C3%A9es-comptes-et-tu)
